### PR TITLE
[AppService] az webapp deployment github-actions add/remove: new commands

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**
New commands for adding/removing github actions to a webapp
- `az webapp deployment github-actions add`
- `az webapp deployment github-actions remove`

**Testing Guide**
Example usage:

- Add:

image

Navigate to https://github.com/login/device to enter code
image

Authorize after entering in code
image

- Remove:
image

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py
Comment (JasonFreeberg) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:4549]: Fill --> Filling

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py
Comment (JasonFreeberg) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:4587]: Add --> Adding

**Review by JasonFreeberg — CHANGES_REQUESTED** [review]
I added two comments about the logging to use a consistent verb tense. 

Can someone explain what's happening with the browser auth? From what I see, the CLI command will open the browser to start the auth flow. Once the user does that, is a PAT token created and placed on the user's dev box? Just trying to understand the auth flow better.

**Comment by yonzhan** [comment]
@Juliehzl to review

**Comment by calvinsID** [comment]
> I added two comments about the logging to use a consistent verb tense.
> 
> Can someone explain what's happening with the browser auth? From what I see, the CLI command will open the browser to start the auth flow. Once the user does that, is a PAT token created and placed on the user's dev box? Just trying to understand the auth flow better.

Thanks for the review Jason! I've updated the description with a few more details, the CLI command will tell you to navigate to the url "github.com/login/device" and enter in the code to authorize us. We will poll and wait up to 14 minutes for the user to do this, and once the user does, we will receive the PAT token from polling and use that token in the command. The token won't be stored anywhere so the user will have to do this process everytime unless they supply the PAT token. We can add a command to just retrieve the PAT token in the future, but I wasn't sure of where to put/how to name the command

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1725]:
```suggestion
    short-summary: Add a GitHub Actions workflow file to the specified repository. The workflow will build and deploy your app to the specified webapp.
```

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1737]:
```suggestion
    short-summary: Remove and disconnects the GitHub Actions workflow file from the specified repository.
```

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:422]: If this flag will take effect when specified, recommend to use `action='store_true'`.

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:426]: If you want to make it take effect when specified, use `action='store_true`

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:250]: what about adding preview tag for one or two sprint to test the functionality for new commands?

File: src/azure-cli/setup.py
Comment (Juliehzl) [inline src/azure-cli/setup.py:142]: please also add in requirementxxxx files.

**Comment by Juliehzl** [comment]
Hi @calvinsID, before merge, please also get approval from other code owners and @JasonFreeberg as he is requesting change,

File: src/azure-cli/setup.py
Comment (calvinsID) [reply src/azure-cli/setup.py:142]: It's already added in requirementsxxx files, so I just added here as well

Comment (jiasli) [reply src/azure-cli/setup.py:143]: Why do we need to pin `PyNaCL`?

Comment (calvinsID) [reply src/azure-cli/setup.py]: unpinned

File: src/azure-cli/setup.py
Comment (jiasli) [inline src/azure-cli/setup.py:142]: Could you unpin `PyGithub` (#18548, #17826)?

Comment (calvinsID) [reply src/azure-cli/setup.py]: Unpinned in `setup.py`. Wondering why in requirements.txt all are pinned?

Comment (jiasli) [reply src/azure-cli/setup.py]: `requirements.txt` is used for building [official packages](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) like msi, deb, rpm. It is usually not possible to be used together with other python libraries in the same environment. That's why we want to pin the version. 

Also, you may check the `pip` doc on `requirements.txt`: https://pip.pypa.io/en/stable/user_guide/#requirements-files

> Requirements files are used to hold the result from pip freeze for the purpose of achieving repeatable installations. In this case, your requirement file contains a pinned version of everything that was installed when pip freeze was run.

**Review by panchagnula — APPROVED** [review]
shipIt

**Review by JasonFreeberg — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1732]: it would be better if you can remove the spaces here to align with other help entries

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1744]: the same

File: src/azure-cli/setup.py
Comment (Juliehzl) [inline src/azure-cli/setup.py:143]: you also need to add dependencies in requirementxxx.txt

File: src/azure-cli/azure/cli/command_modules/appservice/utils.py
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/appservice/utils.py:82]: recommend to use the exception type in azclierror, e.g.
https://github.com/Azure/azure-cli/blob/3eea7fd773722b7a3c40b8115f29a3bad39b70bc/src/azure-cli-core/azure/cli/core/azclierror.py#L142-L144

File: src/azure-cli/setup.py
Comment (calvinsID) [reply src/azure-cli/setup.py:143]: These already existed in requirementxxx.txt

**Comment by panchagnula** [comment]
@Juliehzl can you review the lashes updates on this one & approve & merge?

**Review by Juliehzl — APPROVED** [review]

**Pull request merged**
